### ISSUE TITLE
Rename repository to symengine.f

### ISSUE THREAD

**Issue by rikardn** [symengine/symengine.f90#7]
`symengine.f90` signals bindings to an old fortran standard. Perhaps we could rename it to `symengine.f` to avoid the fortran version in the name?

**Comment by ivan-pi** [comment]
Unfortunately most active Fortran practitioners today (still) associate .f with the old-style F77 source files using fixed format, even if this is unrelated with the actual Fortran standard. I would recommend reading the blog post by Steve Lionel to learn more about the (mis)usage of Fortran file extensions (https://stevelionel.com/drfortran/2013/01/11/doctor-fortran-in-source-form-just-wants-to-be-free/). Also see https://github.com/fortran-lang/fpm/issues/250 for a summary of the various file extensions recognized by compilers and how they are processed by default.

That being said, there was a long discussion about trying to reclaim .f as the default Fortran extensions in one of the fpm issues: https://github.com/fortran-lang/fpm/issues/363. I believe @certik was be in favor of this. I agree _symengine-dot-eff_ sounds more catchy than _symengine-dot-eff-ninety_.

**Comment by certik** [comment]
Yes, we should rename to `symengine.f`. As @ivan-pi correctly summarized, historically `.f` meant `f77`, but we have proposed in https://github.com/fortran-lang/fpm/issues/363 to reclaim `.f` to just mean any Fortran (from f77 to modern Fortran, using free form by default). And in this case, we are talking about the name of the github repository, not the file extensions, so there is almost no issue. 

Let me try to get some support for this at https://github.com/fortran-lang/fpm/issues/363 in the Fortran community, but as far as I am concerned, `symengine.f` is better:

* https://github.com/fortran-lang/fpm/issues/363#issuecomment-886168107

**Comment by rikardn** [comment]
Thanks @ivan-pi and @certik for taking time to explain. I naïvely assumed that ```.f``` didn't carry any Fortran standard meaning. For the little Fortran programming I do professionally myself I mostly use `.f08` to get good compiler defaults. I don't have a strong opinion on the matter myself so I'll leave the decision to you guys.
